### PR TITLE
HPCC-32470 Document OUTPUT and BUILD support for PLANE option

### PR DESCRIPTION
**Pull request by JamesDeFabia**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

https://github.com/JamesDeFabia/github-action-dev-build/actions/runs/10707996179

### CODE REVIEW

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-BUILD.xml [hunk 1]
```diff
@@ -256,17 +258,34 @@
...
-              <emphasis>indexfile</emphasis> to. The names must be listed as
-              they appear on the ECL Watch Activity page or returned by the
-              Std.System.Thorlib.Group() function, optionally with square
-              brackets containing a comma-delimited list of node-numbers
-              (1-based) and/or ranges (specified with a dash, as in n-m) to
-              indicate the specific set of nodes to write to.</entry>
+              names of the clusters or planes to write the
+              <emphasis>indexfile</emphasis> to. The cluster names must be
+              listed as they appear on the ECL Watch Activity page or returned
+              by the Std.System.Thorlib.Group() function, optionally with
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-BUILD.xml:281]: >The cluster names must be listed as they appear on the ECL Watch Activity page or returned by the Std.System.Thorlib.Group() function.

Not true in containerized. Planes are independent and there may be many (and I am not sure if they are listed in EclWatch anywhere at the moment - I've opened JIRA: https://hpccsystems.atlassian.net/browse/HPCC-32624).
And Thorlib.Group() makes no sense in containerized - there is no fixed group (of nodes) in cloud.

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-BUILD.xml [hunk 2]
```diff
@@ -256,17 +258,34 @@
...
+              square brackets containing a comma-delimited list of
+              node-numbers (1-based) and/or ranges (specified with a dash, as
+              in n-m) to indicate the specific set of nodes to write to.
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-BUILD.xml:284]: I think this description would benefit from clearly separating out the BM/Cluster meaning/semantics, to the Containerized/Plane semantics.

At 1st glance, the above sounds like it generically applies to both, but Std.System.Thorlib.Group() makes no sense in containerized, nor does the optional node-number functionality.

Not true in containerized. Planes are independent and there may be many (and I am not sure if they are listed in EclWatch anywhere at the moment - I've opened JIRA: https://hpccsystems.atlassian.net/browse/HPCC-32624).
And Thorlib.Group() makes no sense in containerized - there is no fixed group (of nodes) in cloud.

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml [hunk 3]
```diff
@@ -473,17 +475,32 @@
...
 
+            <row>
+              <entry><emphasis role="bold">PLANE</emphasis></entry>
+
+              <entry>Optional. Specifies writing the file to the specified
+              list of target planes. If omitted, the file is written to the
+              default plane for the cluster on which the workunit executes.
+              The number of physical file parts written to disk is always
+              determined by the number of nodes in the cluster on which the
+              workunit executes, regardless of the number of nodes on the
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml:486]: see previous comments - i.e. no nodes involved on a data plane..

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml [hunk 4]
```diff
@@ -473,17 +475,32 @@
...
 
               <entry>A comma-delimited list of string constants containing the
-              names of the clusters to write the file to. The names must be
-              listed as they appear on the ECL Watch Activity page or returned
-              by the Std.System.Thorlib.Group() function, optionally with
-              square brackets containing a comma-delimited list of
+              names of the clusters or planes to write the file to. The names
+              must be listed as they appear on the ECL Watch Activity page or
+              returned by the Std.System.Thorlib.Group() function, optionally
+              with square brackets containing a comma-delimited list of
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml:497]: see prev. comments re. clarifying semantics between BM/Cluster and Containerized/Plane

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml [hunk 5]
```diff
@@ -632,17 +651,32 @@ OUTPUT(People(Attr1=FALSE));
...
+            <row>
+              <entry><emphasis role="bold">PLANE</emphasis></entry>
+
+              <entry>Optional. Specifies writing the file to the specified
+              list of target planes. If omitted, the file is written to the
+              default plane for the cluster on which the workunit executes.
+              The number of physical file parts written to disk is always
+              determined by the number of nodes in the cluster on which the
+              workunit executes, regardless of the number of nodes on the
+              target. </entry>
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml:663]: see prev. comments

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml [hunk 6]
```diff
@@ -632,17 +651,32 @@ OUTPUT(People(Attr1=FALSE));
...
 
               <entry>A comma-delimited list of string constants containing the
-              names of the clusters to write the file to. The names must be
-              listed as they appear on the ECL Watch Activity page or returned
-              by the Std.System.Thorlib.Group() function, optionally with
-              square brackets containing a comma-delimited list of
+              names of the clusters or planes to write the file to. The names
+              must be listed as they appear on the ECL Watch Activity page or
+              returned by the Std.System.Thorlib.Group() function, optionally
+              with square brackets containing a comma-delimited list of
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml:673]: see prev. comments

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml [hunk 7]
```diff
@@ -917,17 +953,32 @@ OUTPUT(ds,,'~thor::outdata.csv',CSV(HEADING(FORMAT(STD.Str.ToUpperCase))));
...
+            <row>
+              <entry><emphasis role="bold">PLANE</emphasis></entry>
+
+              <entry>Optional. Specifies writing the file to the specified
+              list of target planes. If omitted, the file is written to the
+              default plane for the cluster on which the workunit executes.
+              The number of physical file parts written to disk is always
+              determined by the number of nodes in the cluster on which the
+              workunit executes, regardless of the number of nodes on the
+              target. </entry>
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml:965]: see prev. comments

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml [hunk 8]
```diff
@@ -917,17 +953,32 @@ OUTPUT(ds,,'~thor::outdata.csv',CSV(HEADING(FORMAT(STD.Str.ToUpperCase))));
...
               <entry>A comma-delimited list of string constants containing the
-              names of the clusters to write the file to. The names must be
-              listed as they appear on the ECL Watch Activity page or returned
-              by the Std.System.Thorlib.Group() function, optionally with
-              square brackets containing a comma-delimited list of
+              names of the clusters or planes to write the file to. The names
+              must be listed as they appear on the ECL Watch Activity page or
+              returned by the Std.System.Thorlib.Group() function, optionally
+              with square brackets containing a comma-delimited list of
               node-numbers (1-based) and/or ranges (specified with a dash, as
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml:976]: see prev. comments

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml [hunk 9]
```diff
@@ -1142,6 +1193,18 @@ OUTPUT(B,,'fred3.xml',XML('MyRow',TRIM,OPT));
...
+            <row>
+              <entry><emphasis role="bold">PLANE</emphasis></entry>
+
+              <entry>Optional. Specifies writing the file to the specified
+              list of target planes. If omitted, the file is written to the
+              default plane for the cluster on which the workunit executes.
+              The number of physical file parts written to disk is always
+              determined by the number of nodes in the cluster on which the
+              workunit executes, regardless of the number of nodes on the
+              target.</entry>
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml:1205]: see prev. comments

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml [hunk 10]
```diff
@@ -1151,8 +1214,11 @@ OUTPUT(B,,'fred3.xml',XML('MyRow',TRIM,OPT));
               by the Std.System.Thorlib.Group() function, optionally with
               square brackets containing a comma-delimited list of
               node-numbers (1-based) and/or ranges (specified with a dash, as
```
Comment (jakesmith) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-OUTPUT.xml:1216]: see prev. comments

**Review by jakesmith — COMMENTED** [review]
@JamesDeFabia - please see comments

**Comment by JamesDeFabia** [comment]
@jakesmith Please have another look. I completely revised my approach based on your feedback.

**Review by jakesmith — APPROVED** [review]
@JamesDeFabia - looks good.

@ghalliday - should this target an earlier version?

**Comment by ghalliday** [comment]
Merged as is - people tend to use the most recent documentation I would have thought.

**Pull request merged**